addons/im_livechat/static/tests/chat_window_patch.test.js

```diff
@@ -72,6 +80,105 @@ test("closing a chat window with no message from admin side unpins it", async ()
     await contains(".o-mail-DiscussSidebarChannel", { count: 0, text: "Partner 2" });
 });
 
+test("Should open active livechat first on Tab key press", async () => {
+    const pyEnv = await startServer();
+    const guestIds = pyEnv["mail.guest"].create([
+        { name: "Visitor 1" },
+        { name: "Visitor 2" },
+        { name: "Visitor 3" },
+        { name: "Visitor 4" },
+        { name: "Visitor 5" },
+    ]);
+    const livechatChannelId = pyEnv["im_livechat.channel"].create({
+        name: "Test",
+        user_ids: [serverState.userId],
+    });
+    const channelIds = pyEnv["discuss.channel"].create(
+        guestIds.map((guestId) => ({
+            channel_type: "livechat",
+            channel_member_ids: [
+                Command.create({ partner_id: serverState.partnerId }),
+                Command.create({ guest_id: guestId }),
+            ],
+            livechat_active: true,
+            livechat_channel_id: livechatChannelId,
+            livechat_operator_id: serverState.partnerId,
+            create_uid: serverState.publicUserId,
+        }))
+    );
+    setupChatHub({ folded: [channelIds[0], channelIds[1], channelIds[2], channelIds[3]] });
+    await start();
+    await click(".o_menu_systray i[aria-label='Messages']");
+    await click(".o-mail-NotificationItem", { text: "Visitor 5" });
+    await contains(".o-mail-ChatWindow", { text: "Visitor 5" });
+    await contains(".o-mail-Composer-input[placeholder='Message Visitor 5…']:focus");
+    await withGuest(guestIds[1], () =>
+        rpc("/mail/message/post", {
+            post_data: {
+                body: "livechat 2",
+                message_type: "comment",
+                subtype_xmlid: "mail.mt_comment",
+            },
+            thread_id: channelIds[1],
+            thread_model: "discuss.channel",
+        })
+    );
+    await withGuest(guestIds[1], () =>
+        rpc("/im_livechat/visitor_leave_session", { channel_id: channelIds[1] })
+    );
+    await withGuest(guestIds[3], () =>
+        rpc("/mail/message/post", {
+            post_data: {
+                body: "livechat 4",
+                message_type: "comment",
+                subtype_xmlid: "mail.mt_comment",
+            },
+            thread_id: channelIds[3],
+            thread_model: "discuss.channel",
+        })
+    );
+    await animationFrame();
+    await press("Tab");
+    await contains(".o-mail-ChatWindow", { count: 2 });
```

Comment:
Would await presence of `.o-mail-Message` in Visitor 4 conversation, rather than `await animationFrame()`, as the awaiting isn't necessarily exactly after a single animation frame. Presence of `.o-mail-Message` (or the badge counter on chat window) is more reliable to tell whether the client code is ready to consider this conversation as more recent.